inithome/tests/conftest.py

```diff
@@ -0,0 +1,80 @@
+"""Fixtures for inithome testing."""
+
+import os
+from dataclasses import dataclass
+from pathlib import Path
+
+import pytest
+from pyfakefs.fake_filesystem import FakeFilesystem, set_gid, set_uid
+
+
+@dataclass
+class User:
+    homedir: Path
+    uid: int
+    gid: int
+
+
+USERS = {
+    "gsamsa": User(homedir=Path("/home/gsamsa"), uid=2247, gid=200),
+    "josephk": User(
+        homedir=Path("/home/j/josephk/nublado"), uid=63928, gid=63928
+    ),
+    "leni": User(homedir=Path("/home/leni"), uid=2000, gid=200),
+    "grubach": User(homedir=Path("/home/grubach"), uid=9942, gid=500),
+    "karl": User(homedir=Path("/home/karl"), uid=1088, gid=500),
+    "huld": User(homedir=Path("/home/huld"), uid=4346, gid=4346),
+    "burstner": User(homedir=Path("/home/burstner"), uid=7304, gid=7304),
+}
```

Comment:
It's always tricky to figure out how to divide this stuff between fixtures and tests, and I don't always do this cleanly (and sometimes it's almost impossible to do this cleanly). But in general I find the pattern where the initial test data is defined in one file and then the tests assume that data and make assertions about it in a different file to be hard to read. I have to keep switching back and forth between the files to understand the relationship between the test and the setup.

I like to either be entirely data driven, so that the input and output data are stored as data in files together so that it's obvious that this input leads to that output, which doesn't work here and would be overkill, or put the test setup data in the same file and ideally the same function as the tests so at least it's easy to see both at the same time.

As such, I'd move all of this except `set_uid`, `set_gid`, and maybe for creating `/home` into the test file rather than putting it in `conftest.py` and restructure this so that the setup is done inline immediately before the test that uses it. In other words, make the data for leni and then do the tests. Make the data for karl and then do those tests. All of that can be inline directly in the test, so then it becomes very obvious what the setup is and then what the test is checking. The test can just hard-code all of the paths and UIDs and GIDs that way rather than setting up a bunch of data structures that are only used in one place.